### DISCUSSION TITLE
What the difference between Weight and Guidance strength?

### DISCUSSION THREAD

**Discussion by zhmou** [Mikubill/sd-webui-controlnet#175] [Q&A] [answered]
As indicated in the title, I wondered.
![image](https://user-images.githubusercontent.com/43105172/219829322-6469e747-1bc5-4301-9c7d-7039da083fd9.png)

**Comment by R-N** [comment, marked as answer]
If I understand correctly,
- Weight is the weight of the controlnet "influence". It's analogous to [prompt attention/emphasis](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Features#attentionemphasis). E.g. (myprompt: 1.2). Although this isn't really a good analogy because setting small attention value doesn't work as you'd expect from weight {[1](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/2157), [2](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/6416)}.
- Guidance is the percentage of total steps the controlnet applies (from the beginning/step 0). It's analogous to [prompt editing/shifting](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Features#prompt-editing). E.g. [myprompt::0.8] (It applies from the beginning until 80% of total steps)

**Reply by zhmou** [reply]
Yes, you're right. Thanks for your kindly answering.
This feature was introduced by commit https://github.com/Mikubill/sd-webui-controlnet/commit/2b36e819a03f06cb8b904c721392622216d62bb0 , it surely works as you said.
